src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/World.kt

```diff
@@ -55,6 +55,12 @@ class World (val settings: WorldSettings) {
         area.borders.remove(wo.id)
         area.food.remove(wo.id)
         area.cells.remove(wo.id)
+
+        if (wo is CellState) {
+            wo.connections.keys.forEach { id ->
+                area.cells[id]!!.connections = area.cells[id]!!.connections.filterKeys { it != wo.id }
```

Comment:
I suppose, it would better to implement that code block like that

>                 area.cells[id]?.let { cellState ->
>                     cellState.connections = cellState.connections.filterKeys { it != wo.id }
>                 }
Cuz we don't throw NPE in worth case.